alphagov/paas-compose-broker
# [#143868179] Walking skeleton of Compose service broker

## What

https://www.pivotaltracker.com/story/show/143868179

This is a 'walking skeleton', so intends to only provide the minimal implementation required to be deployable in its intended context. See commit messages for details.

## How to review

To avoid two people reviewing whether this is registrable as a service broker it can be reviewed in conjuction with the corresponding pull request on paas-cf: https://github.com/alphagov/paas-cf/pull/909

Along with the code review and normal shenanigans you can do this:

* Use the example catalog:

  ```
  cp examples/catalog.json .
  ```

* Deploy to Cloud Foundry:

  ```
  cf push --no-start
  ```

* Set the required environment variables:

  ```
  cf set-env compose-broker USERNAME compose-broker
  cf set-env compose-broker PASSWORD unguessable
  ```

* Start the app:

  ```
  cf start compose-broker
  ```

* Check the catalog endpoint:

  ```
  curl -k -u compose-broker:unguessable https://compose-broker.${DEPLOY_ENV}.dev.cloudpipelineapps.digital/v2/catalog
  ```

## Who

Anyone but me or @combor
